### PR TITLE
Generate index and user docs from tfplugindocs

### PR DESCRIPTION
**Pull request by nmuesch**
Based on PR #798 start generating the user + index documentation files
I noticed a couple of small typos, and fixed those here as well

### CODE REVIEW

**Review by zippolyte — APPROVED** [review]

**Comment by zippolyte** [comment]
I think we should disable prettier on the generated files

**Comment by nmuesch** [comment]
Ha, yeah I had that in #798, but forgot to add here. Adding now

**Review by zippolyte — DISMISSED** [review]

File: docs/resources/user.md [hunk 1]
```diff
@@ -22,33 +25,34 @@ resource "datadog_user" "foo" {
...
--   `is_admin`: (Deprecated) (Optional) Whether the user is an administrator. **Warning**: the corresponding query parameter is ignored by the Datadog API, thus the argument would always trigger an execution plan.
--   `role`: (Deprecated) Role description for user. **Warning**: the corresponding query parameter is ignored by the Datadog API, thus the argument would always trigger an execution plan.
--   `access_role`: (Deprecated) Role description for user. Can be `st` (standard user), `adm` (admin user) or `ro` (read-only user). Default is `st`. `access_role` is ignored for new users created with this resource. New users have to use the `roles` attribute.
+- **email** (String) Email address for user.
 
-## Attributes Reference
+### Optional
 
-The following attributes are exported:
+- **access_role** (String) Role description for user. Can be `st` (standard user), `adm` (admin user) or `ro` (read-only user). Default is `st`. `access_role` is ignored for new users created with this resource. New users have to use the `roles` attribute.
```
Comment (pietdaniel) [inline docs/resources/user.md:36]: should we denote that this field is deprecated?

Comment (nmuesch) [reply docs/resources/user.md:36]: Good note, this PR was autogenerated based on whats defined in the Schema. I just opened a PR to add Deprecated to this field and I can regenerate this once thats merged - https://github.com/DataDog/terraform-provider-datadog/pull/834

**Review by pietdaniel — APPROVED** [review]

**Pull request merged**